### DISCUSSION TITLE
Syntax for negative minimum integers (and related) in range refinements

### DISCUSSION THREAD

**Discussion by fmontesi** [jolie/jolie#526] [Polls]
We support writing ranges like this, meaning that `ranking` must be an integer between `-100` and `50`.

```jolie
type t { ranking:int( ranges( [-100,50] ) ) }
```

Sometimes we want to express ranges that include the upper bound for integers, like:

```jolie
type t { ranking:int( ranges( [-100,*] ) ) }
```

However, we have no syntax (yet!) to express the lower bound, which must be (boringly) inserted manually. Current proposals include `-*` and `*`, giving:

```jolie
type t { ranking:int( ranges( [-*,50] ) ) }
```

```jolie
type t { ranking:int( ranges( [*,50] ) ) }
```

The last one would make `*` ambiguous: its meaning would depend on whether it appears on the left- or right-hand side of a range. Note that this would mean that we could write `[*,*]`.

Which one should we go for? `-*` or `*`?

**Comment by klag** [comment]
Another way to see the choice:
- '-*' means -infinite and * means infinite. Here the  *  represents the symbol of infinite.
- [ * , * ] here the '*' just means _unbounded_
